extensions/packageManager/src/client/Library.tsx

```diff
@@ -277,20 +378,41 @@ const Library: React.FC = () => {
     }
   };
 
+  // return true if the name, description or any of the keywords match the search term
+  const applySearchTerm = (i): boolean => {
+    const term = searchTerm.trim().toLocaleLowerCase();
+    return (
+      i.name.toLowerCase().match(term) ||
+      i.description.toLowerCase().match(term) ||
+      i.keywords.filter((tag) => tag.toLowerCase().match(term)).length
+    );
+  };
+
   const getLibraries = async () => {
     try {
       updateAvailableLibraries(undefined);
       setLoading(true);
-      const response = await getLibraryAPI();
-      updateAvailableLibraries(response.data.available);
-      setRecentlyUsed(response.data.recentlyUsed);
+      if (searchTerm) {
+        const response = await getSearchResults();
+        // if we are searching, but there is not a searchUrl, apply a local filter
+        if (!feeds.find((f) => f.key === feed).searchUrl) {
+          response.data.available = response.data.available.filter(applySearchTerm);
+        }
+        updateAvailableLibraries(response.data.available);
+        setRecentlyUsed(response.data.recentlyUsed);
+      } else {
+        const response = await getLibraryAPI();
+        updateAvailableLibraries(response.data.available);
+        setRecentlyUsed(response.data.recentlyUsed);
+      }
       setLoading(false);
     } catch (err) {
       setApplicationLevelError({
         status: err.response.status,
         message: err.response && err.response.data.message ? err.response.data.message : err,
         summary: strings.libraryError,
       });
+      setLoading(false);
```

Comment:
not: since you have this both at the end of try and catch, it can go to finally block